lib/publiccloud/ec2.pm

```diff
@@ -114,14 +114,38 @@ sub upload_img {
 
     die("Create key-pair failed") unless ($self->create_keypair($self->prefix . time, 'QA_SSH_KEY.pem'));
 
+    # AMI of image to use for helper VM to create/build the image on CSP.
+    # AMI is region specific also we need to use different AMI's for on-demand/BYOS uploads
+    # In case needed combination not covered by this hash c<PUBLIC_CLOUD_EC2_UPLOAD_AMI> variable
+    # will be used. Upload will fail without this variable
+    my $ami_id_hash = {
+      'us-west-1-byos' => 'ami-02b1a98531379c966',
+      'us-west-1' => 'ami-05c558c169cfe8d99',
+      'us-west-2-byos' => 'ami-0b167c7e41ab53cb1',
+      'us-west-2' => 'ami-0174313b5af8423d7',
+      'eu-central-1-byos' => 'ami-09740eee15629d44f',
+      'eu-central-1' => 'ami-09e8a19c9eda495b3'
+    };
+
+    my $ami_id_key = $self->region;
+    $ami_id_key .= '-byos' if $img_name =~ /byos/i;
+
+    my $helper_ami_id= '';
+
+    if (exists($ami_id_hash{$ami_id_key})) {
+      $helper_ami_id= $ami_id_hash{$ami_id_key};
+    }
+    else {
+      $helper_ami_id= get_var('PUBLIC_CLOUD_EC2_UPLOAD_AMI');
+    }
+
+    die('Unable to detect AMI for helper VM') unless (defined($ami_id));
```

Comment:
Should this one be `$helper_ami_id` ? 
If so, maybe `get_required_var` would fit better, and you can drop this `die` call.

Reply:
> Shouldn't this one be `$helper_ami_id` ?

yes you are right 

> If so, maybe `get_required_var` would fit better, and you can drop this `die` call.

no because in this case you will get standard "PUBLIC_CLOUD_EC2_UPLOAD_AMI is not defined" which is not exactly the case why we failing here